app/javascript/packs/restaurantsViewScreen/reastaurantsScreen.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Restaurants from './restaurants';
+import { restaurantsProvider } from '../../services/DataProvider';
+import SelectionFilters from './selectionFilters';
+import RestaurantsGoogleMap from './restaurantsMap';
+
+export default class RestaurantsScreen extends React.Component {
+  state = {
+    restaurants: [],
+    filtered: [],
+    rating: 0,
+    cuisine: 'all',
+    maximumDeliveryTime: 'all',
+  };
+
+  setRestaurants = (restaurants) => {
+    this.setState({ restaurants }, this.filter);
+  };
+
+  componentWillMount () {
+    restaurantsProvider.get('/restaurants')
+      .then(this.setRestaurants)
+      .catch(console.error);
+  }
+
+  filter = () => {
+    const filtered = [...this.state.restaurants].filter(restaurant => {
+      var isRatingEligible = restaurant.restaurant_reviews_metadatum.avarage_score >= parseInt(this.state.rating);
+      var isCuisineEligible = this.state.cuisine == 'all' || this.state.cuisine == restaurant.cuisine.name;
+      var isDeliveryTimeEligible = this.state.maximumDeliveryTime == 'all' || this.state.maximumDeliveryTime >= restaurant.delivery_sla_in_minutes;
+      return isRatingEligible && isCuisineEligible && isDeliveryTimeEligible;
+    });
+
+    this.setState({ filtered });
+  };
+
+  updateFilter = (filterToUpdate) => {
+    this.setState(filterToUpdate, this.filter);
+  };
+
+  render () {
+    return (
+      <div>
+        <SelectionFilters updateFilter={this.updateFilter}/>
```

Comment:
if i'm reading this right, it seems `SelectionFilters` has intimate knowledge of how `RestaurantsScreen` internal state objects are built. or perhaps it's just happens to be they both share the same state structure. at any rate it's very brittle, any small change in the state of `SelectionFilters` will break `ResturantsScreen`